sympy/solvers/ode.py

```diff
@@ -4003,6 +4002,33 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+
+def _check_substitution_type(eq, func):
+    # Check if some derivate of positive order can be substituted as g(x) = f^(order_to_subs) (x)
+    x = func.args[0]
+    f = func.func
+    order_to_subs = 0
+    D = Dummy()
+    for nsubs_try in range(1,ode_order(eq,f(x))):
+        if eq.subs(f(x).diff(x, nsubs_try), D).has(f(x)):
+            break
+        else:
+            order_to_subs += 1
+    if order_to_subs > 0:
+        return {'var':order_to_subs, 'solutions':True}
+
+# Use repeated substitution until we do not have a function independent of derivative   
+def order_reducing_substitution(eq, func, order, match):
```

Comment:
This function should be called `ode_order_reducing_substitution`

Reply:
Why?I mean all other functions are named normally and everywhere I have used this name only.

Reply:
And instead of markdown syntax does dpaste.com works?

Reply:
If the hint is called X then the corresponding function should be called ode_X.

Reply:
If the hint is called X then the corresponding function should be called ode_X.

Reply:
No idea about dpaste.com. You should learn the markdown syntax to use github though

Reply:
# Use repeated substitution until we do not have a function independent of derivative   
* def ode_order_reducing_substitution(eq, func, order, match):

Reply:
""Use repeated substitution until we do not have a function independent of derivative   ""
'''def ode_order_reducing_substitution(eq, func, order, match):'''

Reply:
# Use repeated substitution until we do not have a function independent of derivative   
def ode_order_reducing_substitution(eq, func, order, match):

Reply:
-[x] " Use repeated substitution until we do not have a function independent of derivative  " 
-[x]  def ode_order_reducing_substitution(eq, func, order, match):

Reply:
Sorry, it will take multiple efforts to learn that styling. I will read that doc tomorrow.

Reply:
One thing is that the r""" ....""" is coloring whole of the whole below it orange in editor which happened for quoted text in my editor.I removed the r and the code turned from orange to multi-colored.Strange isn't it?It seems as if the quotes aren't being actually closed even after closing them with """.On the other hand while debugging I could see the function names in the debugging window.I don't know if they were from the all_hints list or somewhere else but this is somewhat strange.